### PR TITLE
Enhancement: Add filter by side for trades 

### PR DESCRIPTION
**Pull request by kengio**

### CODE REVIEW

File: app/api/v2/market/trades.rb [hunk 1]
```diff
@@ -22,6 +22,8 @@ class Trades < Grape::API
           current_user
             .trades
             .order(order_param)
+            .tap { |q| q.where!('(taker_id = ? AND taker_type = "sell") OR (maker_id = ? AND taker_type = "buy")', current_user.id, current_user.id) if params[:type] == 'sell' }
+            .tap { |q| q.where!('(taker_id = ? AND taker_type = "buy") OR (maker_id = ? AND taker_type = "sell")', current_user.id, current_user.id) if params[:type] == 'buy' }
```
Comment (chumaknadya) [inline app/api/v2/market/trades.rb]:
```
.tap { |q| q.where!('(taker_id = ? AND taker_type = ?) OR (maker_id = ? AND taker_type = ?)',current_user.id, params[:type], current_user.id, opposite_type_param) if params[:type]
```

And here just create simple function like         
```
def opposite_type_param
     params[:type] == 'buy' ? 'sell' : 'buy'
 end
 ```

Comment (kengio) [reply app/api/v2/market/trades.rb]: I don't know why calling `opposite_type_param` method in where is doesn't work but if I call it directly is working fine.

```ruby
.tap { |q| q.where!('(taker_id = ? AND taker_type = ?) OR (maker_id = ? AND taker_type = ?)', current_user.id, params[:type], current_user.id, params[:type] == 'buy' ? 'sell' : 'buy') if params[:type] }
```

**Review by chumaknadya — APPROVED** [review]

**Comment by calj** [comment]
Let's investigate the performances of the query

**Comment by kengio** [comment]
> Let's investigate the performances of the query

How? I just add new `taker_type` index on traders. Is it enough?

**Pull request merged**